packages/connect-query/src/connect-query-key.test.ts

```diff
@@ -12,65 +12,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Timestamp, toPlainMessage } from "@bufbuild/protobuf";
+import { create } from "@bufbuild/protobuf";
 import { describe, expect, it } from "@jest/globals";
 
 import { createConnectQueryKey } from "./connect-query-key.js";
-import { OperationRequest, SayRequest } from "./gen/eliza_pb.js";
+import { ElizaService, SayRequestSchema } from "./gen/eliza_pb.js";
 import { disableQuery } from "./utils.js";
 
 describe("makeQueryKey", () => {
   const methodDescriptor = {
-    I: SayRequest,
+    input: SayRequestSchema,
     name: "name",
-    service: {
-      typeName: "service.typeName",
-    },
+    parent: ElizaService,
   };
 
   it("makes a query key with input", () => {
     const key = createConnectQueryKey(methodDescriptor, {
       sentence: "someValue",
     });
     expect(key).toStrictEqual([
-      "service.typeName",
+      ElizaService.typeName,
       "name",
-      { sentence: "someValue" },
+      create(SayRequestSchema, { sentence: "someValue" }),
```

Comment:
This is interesting. I wonder if it's not very useful to store the full `$typeName` in the key. Not that it's a problem really, just feels unnecessary.

Reply:
It isn't necessary to store the type name. But it is necessary to handle default values of fields with implicit presence. For example:

```proto
syntax = "proto3";
message M {
  uint32 field = 1;
}
```

The field has an intrinsic default value `0`, and we have to make sure that the following two calls yield the same query key:

```
createConnectQueryKey(foo, {});
createConnectQueryKey(foo, {field: 0});
```

Otherwise, two queries with exactly the same request payload would have different keys.

`create()` sets the property to `0` in both cases, same as the `new` call did previously. Previously, we converted the class to a plain object again, so that it can be hashed deterministically by the [default hash function](https://github.com/TanStack/query/blob/9da00d610ab437ad1da6854c9a5697874ebebb7e/packages/query-core/src/utils.ts#L205), or by stable-hash from our [default options](https://github.com/connectrpc/connect-query-es/blob/6fcd101896b8ba344f174fe9a46a690554e47206/packages/connect-query/src/default-options.ts#L16) (necessary if bigint is used).

In v2, proto3 messages are plain objects. What's missing here is support for proto2 and editions, which will use the prototype chain, and not hash deterministically.

I think we should take care of that separately, with added test coverage. I don't think we need stable-hash and can still play nice with default hash function, and also resolve #375.

Reply:
Yeah that definitely makes sense. Leaving the type name works for me.